porcelain/sectorbuilder.go

```diff
@@ -45,19 +78,9 @@ func SealNow(ctx context.Context, plumbing sbPlumbing) error {
 		return errors.Wrap(err, "could not retrieved staged sectors")
 	}
 
-	// if no sectors are staged, add a 1 byte piece to ensure at least one seal
+	// nothing to do if no sectors are staged
 	if len(stagedSectors) == 0 {
-		data := []byte{0}
-		hash, err := multihash.Sum(data, multihash.SHA2_256, -1)
-		if err != nil {
-			return errors.Wrap(err, "could not create cid for piece")
-		}
-		pieceRef := cid.NewCidV1(cid.DagCBOR, hash)
-		_, err = plumbing.SectorBuilder().AddPiece(ctx, pieceRef, 1, bytes.NewReader(data))
-		if err != nil {
-			return errors.Wrap(err, "could not add piece to trigger sealing")
-		}
-
+		return nil
```

Comment:
IMO this is unnecessary. If the sector builder can't handle it, leave a comment explaining that and file an issue on it.

Reply:
This should really error out here. If the miner has asked to seal and we can't because we have no staged sectors, it should not fail silently.